File: vortexdbg-api/src/main/java/com/vortexdbg/hook/hookzz/InstrumentCallback.java
package com.vortexdbg.hook.hookzz;

import com.vortexdbg.Emulator;
import com.vortexdbg.arm.context.RegisterContext;

public abstract class InstrumentCallback<T extends RegisterContext> {

    public abstract void dbiCall(Emulator<?> emulator, T ctx, HookEntryInfo info);

}

File: vortexdbg-api/src/main/kotlin/com/vortexdbg/hook/hookzz/InstrumentCallback.kt
package com.vortexdbg.hook.hookzz

import com.vortexdbg.Emulator
import com.vortexdbg.arm.context.RegisterContext

abstract class InstrumentCallback<T : RegisterContext> {

    abstract fun dbiCall(emulator: Emulator<*>, ctx: T, info: HookEntryInfo)

}
